### PR TITLE
Ivan/pattern decorator

### PR DESCRIPTION
**Pull request by ZaichenkoIvan**
Визначити специфікації класів для подання графічних маніпуляторів
геометричних властивостей(положення, розмір) у редакторі векторної
графіки.

### CODE REVIEW

**Review by ZaichenkoVasia — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -5,10 +5,11 @@ Links:
 * [Adapter](src/myn/patterns/adapter)
 * [Builder](src/myn/patterns/builder)
 * [Composite](src/myn/patterns/composite) 
+* [Decorator](src/myn/patterns/decorator)
```
Comment (hunter04d) [inline README.md]: I like the idea behind sorting alphabetically

**Review by hunter04d — COMMENTED** [review]
1. You can move create an abstract super class for all of the decorators so your code does not repeat itself. However one can argue that that improves clarity in this case.

2. I am a bit confused. You pass the `AbstractShapeRedactor ` into the constructor to every decorator and also to the `#draw()` method itself? So the Redactor and the shape is all in one? Then shape can draw itself? I am very confused about the logic behind all of this. Please collaborate on this.

File: README.md (hunk 1 above)
Comment (ZaichenkoIvan) [reply README.md]: thx)

File: src/myn/patterns/decorator/GraphicRedactorDecorator/AbstractShape.java [hunk 2]
```diff
@@ -2,26 +2,30 @@
 
 /**
  * @author Zaichenko Ivan
- * Abstract class AbstractShapeRedactor for work of shape
+ * Abstract class AbstractShape for work of shape
  */
-public abstract class AbstractShapeRedactor {
+public abstract class AbstractShape {
```
Comment (hunter04d) [inline src/myn/patterns/decorator/GraphicRedactorDecorator/AbstractShape.java]: Do you realize that every decorator has a copy of `x`, `y`, and `size` that they don't use?
And `Shape` has `shapeClone` that it does not use.
That's *24 bytes* per decorator (if memory serves me right) and *8 bytes* per Shape.

**Review by hunter04d — COMMENTED** [review]
To make the code more readable and potentially waste less space I suggest the following change to class hierarchy (also please see the decorator pattern). 
image

**Review by hunter04d — APPROVED** [review]

**Review by ZaichenkoIvan — COMMENTED** [review]
I firstly think about it.
But I dont know, why I didnt realise this...

**Review by ZaichenkoVasia — APPROVED** [review]

**Pull request merged**